BinaryObjectScanner/Protection/CopyKiller.cs

```diff
@@ -1,32 +1,65 @@
+    /// 
+    /// CopyKiller website: https://web.archive.org/web/20030312200712/http://www.webstylerzone.com/CopyKiller/index.htm
+    /// Version 3.62 Installer: https://web.archive.org/web/20031130192048/http://www.webstylerzone.com/Downloads/Brennertools/CopyKiller-Setup.exe
+    /// Version 3.64 Installer: https://web.archive.org/web/20060524220845/http://download.webstylerzone.com:80/exe/CopyKiller-Setup.exe
+    /// Version 3.99 Installer: https://web.archive.org/web/20060524220845/http://download.webstylerzone.com:80/exe/CopyKiller-Setup.exe
+    /// Version 3.99a Installer: https://web.archive.org/web/20070721070138/http://www.webstylerzone.com/Downloads/exe/CopyKiller-Setup.exe
+    /// Version 3.99a Portable: https://web.archive.org/web/20070721070214/http://www.webstylerzone.com/Downloads/zip/CopyKiller.zip
+    /// 
+    /// TODO: Add support for the developer's EXE obfuscator, "EXEShield Deluxe".
+    /// Samples include CopyKiller (Versions 3.64 & 3.99a) and SafeDiscScanner (Version 0.16) (https://archive.org/details/safediscscanner-0.16-webstylerzone-from-unofficial-source).
+    /// 
+    /// </summary>
+    public class CopyKiller : IPathCheck, IPortableExecutableCheck
     {
         /// <inheritdoc/>
-        public string? CheckContents(string file, byte[] fileContent, bool includeDebug)
+        public string? CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
         {
-            // TODO: Obtain a sample to find where this string is in a typical executable
-            if (includeDebug)
-            {
-                var contentMatchSets = new List<ContentMatchSet>
-                {
-                    // Tom Commander
-                    new(new byte?[]
-                    {
-                        0x54, 0x6F, 0x6D, 0x20, 0x43, 0x6F, 0x6D, 0x6D,
-                        0x61, 0x6E, 0x64, 0x65, 0x72
-                    }, "CopyKiller"),
-                };
+            // TODO: Figure out how to differentiate between V3.99 and V3.99a.
+            // Get the sections from the executable, if possible
+            var sections = pex.Model.SectionTable;
+            if (sections == null)
+                return null;
+
+            // TODO: Figure out why this check doesn't work.
+            // Found in ""autorun.exe" in CopyKiller V3.64, V3.99, and V3.99a.
+            var name = pex.ProductName;
+            if (name?.StartsWith("CopyKiller", StringComparison.OrdinalIgnoreCase) == true)
+                return $"CopyKiller V3.64+";
 
-                return MatchUtil.GetFirstMatch(file, fileContent, contentMatchSets, includeDebug);
-            }
+            else if (name?.StartsWith("Solidshield Activation Library", StringComparison.OrdinalIgnoreCase) == true)
+                return $"SolidShield Core.dll {pex.GetInternalVersion()}";
```

Comment:
I assume this was a leftover from debugging.

Reply:
oops